kong/plugins/file-log/handler.lua

```diff
@@ -17,10 +17,6 @@ local S_IROTH = system_constants.S_IROTH()
 local oflags = bit.bor(O_WRONLY, O_CREAT, O_APPEND)
 local mode = bit.bor(S_IRUSR, S_IWUSR, S_IRGRP, S_IROTH)
 
-ffi.cdef[[
-int write(int fd, const void * ptr, int numbytes);
```

Comment:
There is precedent for moving FFI cdefs into this module (we did this when we update the random bytes function to read from urandom). Having said that, I'm not opposed to moving all cdefs into a separate module. Currently we only call `ffi.cdef` in this module. I'm sure this could be standardized and implemented robustly as part of a plugins API. Is that the scope here? I can create a separate module like `kong/tools/cdefs.lua` or some such if that would help assuage your concerns here.

Reply:
Could we just `ffi.cdef` the `write` function is both places and avoid this issue altogether? Makes the plugin more self-contained (and we'll revisit the topic of exported functionality anyway when we do the Plugin API and do the eventual cleanup if it's the case). AFAIK (and my quick tests showed) LuaJIT doesn't have a problem with duplicated cdefs.

Reply:
@hishamhm no, duplicating the cdef is illegal. This is why we moved the `open()` and `close()` defs in https://github.com/Kong/kong/pull/2536

Reply:
Ouch, is it? I didn't find it in the docs and didn't seem to bark on my tests, but I know all security guarantees are off the window when it comes to the FFI. 

Still, this looks real odd: this means when using two third-party FFI-based modules, one can only hope that they are not using the same C functions internally? Ew.

Reply:
Ok, I must be sniffing glue. calling `cdef` in the module scope lets `require` take care of not executing the def multiple times. I swear I didn't have the problem before... ill revert this change and do some more digging

Reply:
> calling cdef in the module scope lets require take care of not executing the def multiple times. 

What I meant was not two modules calling the same FFI-using module, but two modules calling two different third-party FFI modules which happen to declare `open` (`require` won't protect that).